### PR TITLE
Upgrade from .NET Core 3.0 to .NET Core 3.1 in tests and CLI

### PR DESCRIPTION
**Pull request by DaveTryon**
#### Describe the change

According to https://dotnet.microsoft.com/platform/support/policy/dotnet-core, .NET Core 3.1 is the LTS version, and .NET Core 3.0 (which we use for tests and CLI) isn't officially supported anymore. We originally had to use .NET Core because note all build environments supported .NET Core 3.1. That has now been fixed, so this upgrades us to .NET Core 3.1 for tests and the CLI. The CLI installer had to be modified to use an updated version check and to reflect the fact that the set of created files are slightly different (no more `netcoreapp20`, `netcoreapp21` or `netstandard20` runtime folders, and we ship fewer assemblies because we leverage the versions in the runtime.

Testing included signed builds, installing the CLI MSI, validating the install location, validating that apps can be tested, and that the resulting a11ytest file loads successfully into AIWin

> Note: After the PR has been created, certain checks will be kicked off. All of these checks must pass before a merge.

### CODE REVIEW

File: src/CLI_Installer/Product.wxs [hunk 1]
```diff
@@ -20,41 +20,35 @@ Licensed under the MIT license. See LICENSE file in the project root for full li
...
       </DirectorySearch>
     </Property>
 
     <MediaTemplate EmbedCab="yes" />
 
     <Feature Id="ProductFeature" Title="Axe.Windows CLI" Level="1">
       <ComponentGroupRef Id="ProductComponents" />
-      <ComponentGroupRef Id="NetCoreApp20Components" />
-      <ComponentGroupRef Id="NetCoreApp21Components" />
       <ComponentGroupRef Id="NetCoreApp30Components" />
```
Comment (RobGallo) [inline src/CLI_Installer/Product.wxs:37]: Is there a need for "NetCoreApp31Components"?

File: src/CLI_Installer/Product.wxs [hunk 2]
```diff
@@ -20,41 +20,35 @@ Licensed under the MIT license. See LICENSE file in the project root for full li
...
       <Directory Id="ProgramFilesFolder" >
         <Directory Id="AxeWindowsCLIFolder" Name ="AxeWindowsCLI" >
-          <Directory Id="INSTALLFOLDER" Name="1.0" >
+          <Directory Id="INSTALLFOLDER" Name="$(var.SemVer)" >
             <Directory Id="RuntimesFolder" Name="runtimes" >
               <Directory Id="WinFolder" Name="win" >
                 <Directory Id="LibFolder" Name="lib" >
-                  <Directory Id="NetCoreApp20Folder" Name="netcoreapp2.0" />
-                  <Directory Id="NetCoreApp21Folder" Name="netcoreapp2.1" />
                   <Directory Id="NetCoreApp30Folder" Name="netcoreapp3.0" />
```
Comment (RobGallo) [inline src/CLI_Installer/Product.wxs:51]: No "NetCoreApp31Folder"?

Comment (DaveTryon) [reply src/CLI_Installer/Product.wxs:51]: There's no `netcoreapp3.1` folder. The `runtimes` tree created by the CLI project includes only `runtimes\unix\lib\netcoreapp3.0` (which we omit since UIA is a Windows-specific technology) and `runtimes\win\lib\netcoreapp3.0`

File: src/CLI_Installer/Product.wxs (hunk 1 above)
Comment (DaveTryon) [reply src/CLI_Installer/Product.wxs:37]: No. Please see my other comment.

File: src/CLI_Installer/Product.wxs [hunk 3]
```diff
@@ -20,41 +20,35 @@ Licensed under the MIT license. See LICENSE file in the project root for full li
...
-    <Condition Message="[ProductName] requires .NET Core Runtime 3.0 or newer. Please visit https://dotnet.microsoft.com/download/dotnet-core">
+    <Condition Message="[ProductName] requires .NET Core Runtime 3.1 or newer. Please visit https://dotnet.microsoft.com/download/dotnet-core">
       <![CDATA[Installed OR NETCORERUNTIMEFOUNDX64 OR NETCORERUNTIMEFOUNDX86]]>
     </Condition>
 
     <Directory Id="TARGETDIR" Name="SourceDir" >
       <Directory Id="ProgramFilesFolder" >
         <Directory Id="AxeWindowsCLIFolder" Name ="AxeWindowsCLI" >
-          <Directory Id="INSTALLFOLDER" Name="1.0" >
+          <Directory Id="INSTALLFOLDER" Name="$(var.SemVer)" >
```
Comment (DaveTryon) [inline src/CLI_Installer/Product.wxs:47]: This isn't really related to the .NET Core 3.1 changes--could be a separate PR if desired
